components/org.wso2.carbon.identity.api.server.userstore/org.wso2.carbon.identity.api.server.userstore.v1/src/main/java/org/wso2/carbon/identity/api/server/userstore/v1/core/ServerUserStoreService.java

```diff
@@ -987,4 +989,77 @@ private String extractPropertyName(String pathProp) {
+
+        Set<String> classNames;
+        String userStoreName = getUserStoreType(base64URLDecodeId(typeId));
+        if (StringUtils.isBlank(userStoreName)) {
+            throw handleException(Response.Status.BAD_REQUEST, UserStoreConstants.ErrorMessage.
+                    ERROR_CODE_INVALID_INPUT);
+        }
+        try {
+            UserStoreConfigService userStoreConfigService = UserStoreConfigServiceHolder.getInstance()
+                    .getUserStoreConfigService();
+            classNames = userStoreConfigService.getAvailableUserStoreClasses();
+            if (CollectionUtils.isEmpty(classNames) ||
+                    !classNames.contains(userStoreName)) {
+                throw handleException(Response.Status.NOT_FOUND, UserStoreConstants.ErrorMessage.
+                        ERROR_CODE_NOT_FOUND);
+            }
+        } catch (IdentityUserStoreMgtException e) {
+            UserStoreConstants.ErrorMessage errorEnum =
+                    UserStoreConstants.ErrorMessage.ERROR_CODE_ERROR_RETRIEVING_USER_STORE;
+            throw handleIdentityUserStoreMgtException(e, errorEnum);
+        }
+        UserStoreAttributeMapping userStoreAttributeMapping = new UserStoreAttributeMapping();
+        List<UserStoreAttributeDO> attributeMappings = getAttributeMappings(userStoreName);
+        if (CollectionUtils.isEmpty(attributeMappings)) {
+            throw handleException(Response.Status.NOT_FOUND, UserStoreConstants.ErrorMessage.ERROR_CODE_NOT_FOUND);
+        }
+        userStoreAttributeMapping = userStoreAttributeMapping.attributeMapping(attributeMappings).typeId(typeId);
+        try {
+            boolean isLocalUserStore = UserStoreManagerRegistry.isLocalUserStore(userStoreName);
+            userStoreAttributeMapping.setIsLocalUserStore(isLocalUserStore);
+        } catch (UserStoreException e) {
+            LOG.error(String.format("Userstore type is not found for %s", userStoreName), e);
+        }
+        return userStoreAttributeMapping;
+    }
+
+    /**
+     * Get user store attribute mappings for a given user store typeId.
+     *
+     * @param userStoreName String user store name (base64 decoded user store id).
+     * @return List of user store attribute mappings for the given typeId.
+     */
+    private List<UserStoreAttributeDO> getAttributeMappings(String userStoreName) {
+
+        UserStoreConfigService userStoreConfigService = UserStoreConfigServiceHolder.getInstance().
+                getUserStoreConfigService();
+        try {
+            UserStoreAttributeMappings userStoreAttributeMappings = userStoreConfigService.
+                    getUserStoreAttributeMappings();
+            try {
+
+                Map<String, UserStoreAttributeDO> mapping = userStoreAttributeMappings
+                        .getUserStoreAttrMappings(userStoreName);
+                return new ArrayList<>(mapping.values());
+
+            } catch (NullPointerException e) {
```

Comment:
handle null objects where necessary instead of catching NPE.

Reply:
fixed